File: js/Prime.js
var Prime = (function() {
	var currentPrime = 2,
		primeCount = 2;
		
	return {
		//primality test from wikipedia
		// https://en.wikipedia.org/wiki/Primality_test#Python_implementation
		isPrime: function(n) {
			if (n < 2) return false;
			if (n === 2 || n === 3) return true;
			if (n % 2 === 0 || n % 3 === 0) return false;
			var largestDivisor = Math.round(Math.sqrt(n));
			for (var i = 5; i <= largestDivisor; i+=6) {
				if (n % i === 0 || n % (i + 2) === 0) return false;
			}
			return true;

		},
		currentPrime: function() {
			return currentPrime;
		},
		nextPrime: function() {
			primeCount++;
			while (!Prime.isPrime(primeCount)) primeCount++;
			currentPrime = primeCount;
			return currentPrime;
		}
	};

}());

File: vb/PrimeConjectures/Prime.vb
Public Class Prime
    ' primality test from wikipedia
    ' https://en.wikipedia.org/wiki/Primality_test#Python_implementation
    Public Shared Function isPrime(ByVal n) As Boolean
        If n < 2 Then
            Return False
        End If
        If n = 2 Then
            Return True
        End If
        If n Mod 2 = 0 Or n Mod 3 = 0 Then
            Return False
        End If

        Dim largestDivisor As Integer = Math.Floor(Math.Sqrt(n))
        For i As Integer = 5 To largestDivisor Step 6
            If n Mod i = 0 Or n Mod (i + 2) = 0 Then
                Return False
            End If
        Next
        Return True
    End Function
End Class